bundles/org.openhab.binding.sensibo/src/main/java/org/openhab/binding/sensibo/internal/handler/SensiboAccountHandler.java

```diff
@@ -0,0 +1,333 @@
+        try {
+            config = loadConfigSafely();
+            logger.debug("Initializing Sensibo Account bridge using config {}", config);
+            model = refreshModel();
+            updateStatus(ThingStatus.ONLINE);
+            initPolling();
+            logger.debug("Initialization of Sensibo account completed successfully for {}", config);
+        } catch (final SensiboConfigurationException e) {
+            logger.info("Error initializing Sensibo data: {}", e.getMessage());
+            model = new SensiboModel(0); // Empty model
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Error fetching initial data: " + e.getMessage());
+        } catch (final SensiboException e) {
+            logger.info("Error initializing Sensibo data: {}", e.getMessage());
+            model = new SensiboModel(0); // Empty model
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error fetching initial data: " + e.getMessage());
+            // Reschedule init
+            scheduler.schedule(this::initializeInternal, 30, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        stopPolling();
+        super.dispose();
+    }
+
+    /**
+     * starts this things polling future
+     */
+    private void initPolling() {
+        stopPolling();
+        statusFuture = scheduler.scheduleWithFixedDelay(this::updateModelFromServerAndUpdateThingStatus,
+                config.refreshInterval, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    protected SensiboModel refreshModel() throws SensiboException {
+        final SensiboModel updatedModel = new SensiboModel(System.currentTimeMillis());
+
+        final GetPodsRequest getPodsRequest = new GetPodsRequest();
+        final List<PodDTO> pods = sendRequest(buildRequest(getPodsRequest), getPodsRequest,
+                new TypeToken<ArrayList<PodDTO>>() {
+                }.getType());
+
+        for (final PodDTO pod : pods) {
+            final GetPodsDetailsRequest getPodsDetailsRequest = new GetPodsDetailsRequest(pod.id);
+
+            final PodDetailsDTO podDetails = sendRequest(buildGetPodDetailsRequest(getPodsDetailsRequest),
+                    getPodsDetailsRequest, new TypeToken<PodDetailsDTO>() {
+                    }.getType());
+
+            updatedModel.addPod(new SensiboSky(podDetails));
+        }
+
+        return updatedModel;
+    }
+
+    private <T> T sendRequest(final Request request, final AbstractRequest req, final Type responseType)
+            throws SensiboException {
+        try {
+            final ContentResponse contentResponse = request.send();
+            final String responseJson = contentResponse.getContentAsString();
+            if (contentResponse.getStatus() == HttpStatus.OK_200) {
+                final JsonObject o = jsonParser.parse(responseJson).getAsJsonObject();
```

Comment:
You can't use your gson parser instead?

Reply:
I could, but it would require quite a bit of GSON converter code in order to parse the varying payload.  IMHO I'd rather use the effort creating new functionality than tiny optimizations.